cmd/finch/nerdctl_remote.go

```diff
@@ -340,6 +340,11 @@ func (nc *nerdctlCommand) run(cmdName string, args []string) error {
 		return inspectContainerOutputHandler(cmd)
 	}
 
+	if cmdName == "compose" && nc.fc.DockerCompat && slices.Contains(runArgs, "version") {
+		cmd := nc.ncc.Create(runArgs...)
+		return handleDockerCompatComposeVersion(cmd)
+	}
+
```

Comment:
nit: extract this entire block into a a function and put it in `nerdctl.go`

Reply:
i thought about it but was a bit on the fence as this is more of logical thing but we can move it out.

Reply:
thinking about it, it creates the issue of return statements so if the above portion is true we would need to account the handling of return here vs later. that makes it need to add additional logic in the main code either way

Reply:
nvm i can refactor that.